README.rst

```diff
@@ -0,0 +1,453 @@
+Health| |Dependency Status|
+
+--------------
+
+If you want to create an alternative twitter search portal, the only way
+would be to use the official twitter API to retrieve Tweets. But that
+interface needs an OAuth account and it makes your search portal
+completely dependent on Twitters goodwill. The alternative is, to scrape
+the tweets from the twitter html search result pages, but Twitter may
+still lock you out on your IP address. To circumvent this, you need many
+clients accessing twitter to scrape search results. This makes it
+neccessary to create a distributed peer-to-peer network of twitter
+scrapers which can all organize, store and index tweets. This solution
+was created with loklak.
+
+What is Loklak ?
+^^^^^^^^^^^^^^^^
+
+It is a server application which is able to collect messages from
+various sources, including twitter. The server contains a search index
+and a peer-to-peer index sharing interface.
+
+--------------
+
+Why should I use this ?
+^^^^^^^^^^^^^^^^^^^^^^^
+
+If you like to be anonymous when searching things, want to archive
+tweets or messages about specific topics and if you are looking for a
+tool to create statistics about tweet topics, then you may consider
+loklak. With loklak you can do: - collect and store a very, very large
+amount of tweets and similar messages - create your own search engine
+for tweets - omit authentication enforcment for API requests on the
+twitter plattform - share tweets and tweet archives with other loklak
+users - search anonymously on your own search portal - create your own
+tweet search portal or statistical evaluations - use Kibana to analyze
+large amounts of tweets as source for statistical data.
+
+--------------
+
+Documentation of the API and Usage Examples
+===========================================
+
+To use the loklak app, first an object of the loklak type needs to be
+created. Do the following to install the ``pip`` module and add it to
+your ``requirements`` for the application.
+
+``pip install python-loklak-api``
+
+To use the GUI client for API, install ``wxPython`` from
+`here <http://www.wxpython.org/download.php>`__ and then install
+`Gooey <https://github.com/chriskiehl/Gooey>`__ ``pip`` module.
+
+``pip install Gooey``
+
+Then add
+
+::
+
+    from gooey import Gooey
+    @Gooey
+
+before ``def main():`` in ``/bin/loklak``
+
+Loklak once installed, can be used in the application as
+
+``from loklak import Loklak``
+
+To create a loklak object you can assign the ``Loklak()`` object to a
+variable. ``variable = Loklak()``
+
+eg. ``l = Loklak()`` This creates an objects whose backend loklak server
+is ``http://loklak.org/``
+
+| If you want to set this API to use your own server, you can now define
+  it by doing
+| ``l = Loklak('http://192.168.192.5:9000/')`` for example or pass a URL
+  to it as
+| ``l = Loklak('http://loklak-super-cluster.mybluemix.net/')``
+
+Note the trailing ``/`` is important and so is ``http://``
+
+API Documentation
```

Comment:
@sudheesh001 @bhaveshAn Do you think we should have a separate wiki page or an MD file for the API documentation?

Reply:
Yes, I think creating a wiki page for API Docs and then referencing it through `README.rst`
would be nice way to go.

Reply:
The problem with `wiki` pages is that they can quickly go out of sync with the rest of the codebase. Maybe generate a `readthedocs` style of documentation for the project?